test/fast/deposit.test.ts

```diff
@@ -87,6 +87,19 @@ describe("DepositManager", async function() {
         );
     });
     it("should allow depositing 2 leaves in a subtree and merging it", async function() {
+        const getStateFromEvent = (evt: Event) => {
+            if (!evt.args) {
+                throw new Error("DepositQueued event missing args");
+            }
+            const { pubkeyID, tokenID, l2Amount } = evt.args;
+            return State.new(
+                pubkeyID.toNumber(),
+                tokenID.toNumber(),
+                l2Amount.toNumber(),
```

Comment:
l2Amount could be so large that beyonds `Number.MAX_SAFE_INTEGER`, we should keep it as bigNumber.
```suggestion
                l2Amount,
```

Reply:
Good catch, and good to know that part of state needs to allow for very large numbers.